service/storage_proxy.cc

```diff
@@ -6469,7 +6471,37 @@ db::hints::manager& storage_proxy::hints_manager_for(db::write_type type) {
 }
 
 future<> storage_proxy::truncate_blocking(sstring keyspace, sstring cfname, std::optional<std::chrono::milliseconds> timeout_in_ms) {
-    return remote().send_truncate_blocking(std::move(keyspace), std::move(cfname), timeout_in_ms);
+    locator::effective_replication_map_ptr erm;
+    const auto& ks = _db.local().find_keyspace(keyspace);
```

Comment:
This change looks out of scope of the commit titled "added truncate table for tablets". I don't see any change which seems related to adding truncate table.

Reply:
Propagating storage_service dep should be a separate commit. Reimplementing truncate another one.

Reply:
I have separated the commit as suggested.